addons/web/static/src/views/calendar/calendar_common/calendar_common_renderer.js

```diff
@@ -294,7 +294,20 @@ export class CalendarCommonRenderer extends Component {
             }
         }
         if (id) {
-            res.id = this.props.model.records[id].id;
+            const existingRecord = this.props.model.records[id];
+            if (this.props.model.scale === "month" && existingRecord) {
```

Comment:
either existingRecord can be removed or there is a problem below with
res.id = existingRecord.id;